kernel/kernel-api/src/main/java/io/delta/kernel/client/JsonHandler.java

```diff
@@ -52,6 +52,17 @@ public interface JsonHandler
      */
     ColumnarBatch parseJson(ColumnVector jsonStringVector, StructType outputSchema);
 
+    /**
+     * Parse the given {@code schemaString} according to the Delta Protocol
+     * <a href="https://github.com/delta-io/delta/blob/master/PROTOCOL.md#primitive-types">
+     *    serialization rules </a>.
+     *
+     * @param schemaString the JSON formatted schema string to parse
+     * @return the parsed {@link StructType}
+     */
+    // TODO solidify method name
+    StructType parseStructType(String schemaString);
```

Comment:
`deserializeStructType`? no strong opinions. 

`serializedStructType` name can be the counterpart for the other way.

Reply:
```java
deserializeTableSchema(String schemaJson)
```

Reply:
I think we don't want to mention "table schema". But we could do either `deserializeStructType` or `deserializeDeltaSchema`

Reply:
```
deserializeStructType(String structTypeJson)
```

SGTM